cloud/main.js

```diff
@@ -17,6 +17,54 @@ Parse.Cloud.define("hello", function(request, response) {
     response.success(getAliasName());
 });
 
+// Replays messages in a channel
+// Response Payload:
+// [{messageOldest},{messageOlder},{messageNewer},{messageNewest}]
+Parse.Cloud.define("replayForAlias", function(request, response) {
+    var requiredParams = ["count", "subkey", "aliasId"];
+    var params = request.params;
+    checkMissingParams(params, requiredParams, response);
+
+    var Alias = Parse.Object.extend("Alias");
+    var query = new Parse.Query(Alias);
+    query.get(params.aliasId, {
+        success: function(alias) {
+            var chatRoomId = alias.get("chatRoomId");
+            var startTimeToken = request.params.startTimeToken; 
+            var endTimeToken = alias.get("createdAt").getTime() * 10000; 
+
+            if (!startTimeToken) {
+                startTimeToken = new Date().getTime() * 10000; // Start token should be now if no token was used for message replay 
+            }
+    
+            pubnub.replayChannel(request.params.subkey, chatRoomId, startTimeToken, endTimeToken,
+                        request.params.count, function(messages) {
+
+                var endTimeToken = messages.startTimeToken;
+                
+                pubnub.replayChannel(request.params.subkey, alias.get("chatRoomId") + '-pnpres', 
+                        startTimeToken, endTimeToken, 9999, function(presence) {
+
+                    presence.results = presence.results.filter(function(presenceEvent) {
+                        return presenceEvent.action == "state-change"
+                    });
+
+                    response.success({"messageEvents":messages.results,
+                                      "presenceEvents":presence.results,
+                                      "endTimeToken": messages.endTimeToken,
+                                      "startTimeToken": messages.startTimeToken});
+
+                }, function(error) {
```

Comment:
No point in wrapping response.error(). since it's already a one-arg function you can place it here instead of this anonymous function